app.py

```diff
@@ -47,35 +48,45 @@
 image_captions = collections.OrderedDict()
 VALID_EXT = ['png', 'jpg', 'jpeg']
 error_raised = []
+app_cookie = 'max-image-caption-generator-web-app'
 
 
-class MainHandler(web.RequestHandler):
+class BaseHandler(web.RequestHandler):
+    def prepare(self):
+        if not self.get_cookie(app_cookie):
+            user_id = str(uuid.uuid4())
+            self.set_cookie(app_cookie, str(uuid.uuid4()))
```

Comment:
`set_cookie` should set `user_id` right? Here it's generating a new UUID

Reply:
Nice catch, the error here was actually that I was logging user_id then creating a new one `str(uuid.uuid4())` still instead of using it. In this implementation user_id is simply the value of the `max-image-caption-generator-web-app` cookie in their web-browser